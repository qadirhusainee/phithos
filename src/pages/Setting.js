import React, { useState, useEffect } from "react";
import { useDispatch, useSelector } from "react-redux";
import type { FunctionComponent } from "react";
import {
  View,
  Text,
  StyleSheet,
  TouchableOpacity,
  Button,
  Image,
  Switch,
  SafeAreaView,
} from "react-native";

import { updateBaner, updateLoggedIn } from "../redux/actions/auth";
import Baner from "../component/Baner";

const Setting: FunctionComponent = () => {
  const dispatch = useDispatch();
  const showBaner = useSelector((state) => state.auth.showBaner);

  const onSubmit = () => {
    dispatch(updateLoggedIn(false));
  };
  const toggleSwitch = (value) => {
    dispatch(updateBaner(value));
  };

  return (
    <SafeAreaView style={{ flex: 1 }}>
      <Baner />
      <View style={{ flex: 1, justifyContent: "center", alignItems: "center" }}>
        <Text>Toggle Baner</Text>
        <Switch
          trackColor={{ false: "#767577", true: "#81b0ff" }}
          thumbColor={showBaner ? "#f5dd4b" : "#f4f3f4"}
          ios_backgroundColor="#3e3e3e"
          onValueChange={toggleSwitch}
          value={showBaner}
        />
        <Button title="Log out" onPress={onSubmit} />
      </View>
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({});

export default Setting;
