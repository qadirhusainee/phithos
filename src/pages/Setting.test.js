import React from "react";
import renderer from "react-test-renderer";
import { render, fireEvent } from "react-native-testing-library";

import Setting from "./Setting";

jest.mock("react-redux", () => {
  const ActualReactRedux = require.requireActual("react-redux");
  return {
    ...ActualReactRedux,
    useSelector: jest.fn().mockImplementation(() => {
      return { showBaner: true };
    }),
    useDispatch: jest.fn().mockImplementation(() => jest.fn()),
  };
});

describe("Setting", () => {
  it("Matches Snapshot", () => {
    const tree = renderer.create(<Setting />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("Showing Baner", () => {
    const { queryByText } = render(<Setting />);

    const element = queryByText("Emulator");
    expect(element).toBeDefined();
  });
});
