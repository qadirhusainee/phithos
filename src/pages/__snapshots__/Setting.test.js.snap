// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Setting Matches Snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "#f2f2f2",
        "borderColor": "grey",
        "borderWidth": 0.5,
        "height": 80,
        "justifyContent": "center",
        "width": 750,
      }
    }
  >
    <Text
      style={
        Object {
          "fontSize": 20,
          "paddingTop": 5,
        }
      }
    >
      Simulator(IOS)
    </Text>
    <View
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "flex-end",
          "marginRight": 20,
          "width": 750,
        }
      }
    >
      <View
        accessibilityRole="button"
        accessibilityStates={Array []}
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
      >
        <View
          style={
            Array [
              Object {},
            ]
          }
        >
          <Text
            style={
              Array [
                Object {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "padding": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Close
          </Text>
        </View>
      </View>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <Text>
      Toggle Baner
    </Text>
    <RCTSwitch
      accessibilityRole="button"
      onChange={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      onTintColor="#81b0ff"
      style={
        Array [
          Object {
            "height": 31,
            "width": 51,
          },
          Object {
            "backgroundColor": "#3e3e3e",
            "borderRadius": 16,
          },
        ]
      }
      thumbTintColor="#f5dd4b"
      tintColor="#767577"
      value={false}
    />
    <View
      accessibilityRole="button"
      accessibilityStates={Array []}
      accessible={true}
      focusable={true}
      isTVSelectable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "opacity": 1,
        }
      }
    >
      <View
        style={
          Array [
            Object {},
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#007AFF",
                "fontSize": 18,
                "padding": 8,
                "textAlign": "center",
              },
            ]
          }
        >
          Log out
        </Text>
      </View>
    </View>
  </View>
</RCTSafeAreaView>
`;
